### PR TITLE
📈  Add coverage pools apy to homepage

### PR DESCRIPTION
**Pull request by michalsmiarowski**
This is an extension of #481.

Changes:
- adds coverage pools APY to hom epage
- include cov pools APY when calculating the highest APY that we display in text "Stake on Ethereum’s first private computer and earn up to XX% APY. on home page 

image

### CODE REVIEW

File: src/lib/coverage-pool/coverage-pools.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import { Contract, providers } from "ethers"
+import AssetPool from "./abis/AssetPool.json"
+import KeepToken from "./abis/KeepToken.json"
+import RewardsPool from "./abis/RewardsPool.json"
+import { UniswapExchangeService } from "../exchange-api/exchange-service"
+import BigNumber from "bignumber.js"
+
+const NETWORK = 1
+const INFURA_RPC_URL =
+  "https://mainnet.infura.io/v3/5348113af423473a85e22220ef987de8"
```
Comment (r-czajkowski) [inline src/lib/coverage-pool/coverage-pools.js:12]: Could we move it to eg. `lib/web3-config/index.js`?

`lib/web3-config/index.js`
```
export const NETWORK = 1
export const INFURA_RPC_URL =
  "https://mainnet.infura.io/v3/5348113af423473a85e22220ef987de8"
```

We already have the same values in `lib/liquidity-rewards/liquidity-rewards.js`

File: src/lib/coverage-pool/coverage-pools.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import { Token } from "../liquidity-rewards/utils"
```
Comment (r-czajkowski) [inline src/lib/coverage-pool/coverage-pools.js:1]: Let's move `Token` class maybe to `lib/helper` ?

**Review by r-czajkowski — CHANGES_REQUESTED** [review]
Left some minor comments. Overall looks good! 👍

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    abi: jest.fn(),
+  }
+})
+
+jest.mock("../abis/RewardsPool.json", () => {
+  return {
+    ...jest.requireActual("../abis/AssetPool.json"),
+    address: jest
+      .fn()
+      .mockImplementation(() => "0x7bCa8838d96D723487823975FbeA80e87d04f343"),
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Fixed in 2a7b3959590c1b093a73ab90976d161eb48a79bb

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+jest.mock("../abis/KeepToken.json", () => {
+  return {
+    ...jest.requireActual("../abis/AssetPool.json"),
+    networks: {
+      "1": {
+        address: jest
+          .fn()
+          .mockImplementation(
+            () => "0x7bCa8838d96D723487823975FbeA80e87d04f342"
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Fixed in 2a7b3959590c1b093a73ab90976d161eb48a79bb

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    abi: jest.fn(),
+  }
+})
+
+jest.mock("../abis/AssetPool.json", () => {
+  return {
+    ...jest.requireActual("../abis/AssetPool.json"),
+    address: jest
+      .fn()
+      .mockImplementation(() => "0x7bCa8838d96D723487823975FbeA80e87d04f341"),
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Fixed in 2a7b3959590c1b093a73ab90976d161eb48a79bb

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    UniswapExchangeService: jest.fn(),
+  }
+})
+
+jest.mock("../abis/AssetPool.json", () => {
+  return {
+    ...jest.requireActual("../abis/AssetPool.json"),
+    address: jest
+      .fn()
+      .mockImplementation(() => "0x7bCa8838d96D723487823975FbeA80e87d04f341"),
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Mock for Asset Pool was duplicated. Fixed this in 7a332e173c95355b105022b338912144cd54467a

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+    abi: jest.fn(),
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Fixed in 2a7b3959590c1b093a73ab90976d161eb48a79bb

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Fixed in 2a7b3959590c1b093a73ab90976d161eb48a79bb

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+    networks: {
+      "1": {
+        address: jest
+          .fn()
+          .mockImplementation(
+            () => "0x7bCa8838d96D723487823975FbeA80e87d04f342"
+          ),
+      },
+    },
+    abi: jest.fn(),
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Fixed in 2a7b3959590c1b093a73ab90976d161eb48a79bb

File: src/lib/coverage-pool/__tests__/coverage-pools.test.js [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+  }
+})
+
+jest.mock("../abis/RewardsPool.json", () => {
+  return {
+    ...jest.requireActual("../abis/AssetPool.json"),
+    address: jest
+      .fn()
+      .mockImplementation(() => "0x7bCa8838d96D723487823975FbeA80e87d04f343"),
+    abi: jest.fn(),
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/__tests__/coverage-pools.test.js]: Fixed in 2a7b3959590c1b093a73ab90976d161eb48a79bb

File: src/lib/coverage-pool/coverage-pools.js [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+  /**
+   * Returns the `RewardsPool` contract's rate per second with which reward
+   * tokens are unlocked.
+   * @return {Promise<string>} The rate per second with which reward tokens are
+   * unlocked.
+   */
+  rewardPoolRewardRate = async () => {
+    return await this.rewardPoolContract.rewardRate()
+    // return await this.rewardPoolContract.makeCall("rewardRate")
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply src/lib/coverage-pool/coverage-pools.js]: Fixed in a7e2b22f1f77d89b59b65b19c62b87136caadc88

File: src/lib/coverage-pool/coverage-pools.js (hunk 1 above)
Comment (michalsmiarowski) [reply src/lib/coverage-pool/coverage-pools.js]: Great idea! Fixed this in 18b8a6ddb5f45d66bcb99f5408675f3ab95f414b

File: src/lib/coverage-pool/coverage-pools.js (hunk 2 above)
Comment (michalsmiarowski) [reply src/lib/coverage-pool/coverage-pools.js]: Fixed in 987f8f175e8697d904f03c5c2c805f32ce34d7bc

**Pull request merged**